src/simplify_algebra.cpp

```diff
@@ -1035,8 +1035,15 @@ struct find_split_concat
             std::find_if(args.begin(), args.end(), [&](auto i) { return i == splits.front(); });
         if(std::distance(it, args.end()) < splits.size())
             return;
+        // Don't do anything if not all operations are slice
+        // if(not std::all_of(it, it + splits.size(), [](instruction_ref x) { 
+        //     return x->get_operator().name() == "slice";
+        // }))
+            // return;
```

Comment:
Why is this commented? That seems like the correct fix.

Reply:
I wanted to show you both ideas I had.  If this looks right to you, I'll comment it in and delete the other.  To confirm:  this matcher should expect that all of the args to the `concat` operator are slice operations--correct?